src/interface/hooks/use-post-credential-definitions.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * This function is used to create a credential definition
+ */
+import { useCallback, useState } from 'react'
+import post from '../api/helpers/post'
+
+export default function usePostCredentialDefinitions() {
+  const path = '/credential-definitions'
+  const transformData = (retrievedData) =>
+    retrievedData.credential_definition_id
+  const [error, setError] = useState(null)
+  const onStartFetch = useCallback(
+    (fetchOrigin, schemaId, persona, setStoreData) => {
+      const params = {}
+      const createBody = (schemaId, persona) => {
+        const supportRevocation = false
+        const sanitizeTitleWUnderscore = (str) => {
+          const space = new RegExp(' ', 'g')
+          str = str.replace(space, '_')
+          return str
+        }
+
+        const schemaDefName = schemaId.split(':')[2]
+        const schemaDefTagName = sanitizeTitleWUnderscore(schemaDefName)
```

Comment:
```js
schemaDefTagName = schemaDefName.replace(/\s+/g, '_')
```

would reduce the number of lines and I don't think it requires a helper function